cuda/matrix/sellp_kernels.cu

```diff
@@ -160,10 +162,80 @@ GKO_INSTANTIATE_FOR_EACH_VALUE_AND_INDEX_TYPE(
     GKO_DECLARE_SELLP_ADVANCED_SPMV_KERNEL);
 
 
+namespace kernel {
+
+
+template <typename ValueType>
+__global__ __launch_bounds__(cuda_config::warp_size) void initialize_zero_dense(
+    size_type num_rows, size_type num_cols, size_type stride,
+    ValueType *__restrict__ result)
+{
+    const auto tidx_x = threadIdx.x + blockDim.x * blockIdx.x;
+    const auto tidx_y = threadIdx.y + blockDim.y * blockIdx.y;
+    if (tidx_x < num_cols && tidx_y < num_rows) {
+        result[tidx_y * stride + tidx_x] = zero<ValueType>();
+    }
+}
+
+
+template <typename ValueType, typename IndexType>
+__global__ __launch_bounds__(default_block_size) void fill_in_dense(
+    size_type num_rows, size_type num_cols, size_type stride,
+    size_type slice_size, const size_type *__restrict__ slice_lengths,
+    const size_type *__restrict__ slice_sets,
+    const IndexType *__restrict__ col_idxs,
+    const ValueType *__restrict__ values, ValueType *__restrict__ result)
+{
+    const auto slice = blockIdx.x;
+    const auto row = threadIdx.x;
+    const auto global_row = slice * slice_size + row;
+
+    if (global_row < num_rows) {
+        for (auto i = threadIdx.y; i < slice_lengths[slice]; i += blockDim.y) {
+            result[global_row * stride +
+                   col_idxs[(slice_sets[slice] + i) * slice_size + row]] +=
+                values[(slice_sets[slice] + i) * slice_size + row];
+        }
+    }
+}
+
+
+}  // namespace kernel
+
+
 template <typename ValueType, typename IndexType>
-void convert_to_dense(
-    std::shared_ptr<const CudaExecutor> exec, matrix::Dense<ValueType> *result,
-    const matrix::Sellp<ValueType, IndexType> *source) NOT_IMPLEMENTED;
+void convert_to_dense(std::shared_ptr<const CudaExecutor> exec,
+                      matrix::Dense<ValueType> *result,
+                      const matrix::Sellp<ValueType, IndexType> *source)
+{
+    const auto num_rows = source->get_size()[0];
+    const auto num_cols = source->get_size()[1];
+    const auto vals = source->get_const_values();
+    const auto col_idxs = source->get_const_col_idxs();
+    const auto slice_lengths = source->get_const_slice_lengths();
+    const auto slice_sets = source->get_const_slice_sets();
+    const auto slice_size = source->get_slice_size();
+    const auto slice_num =
+        ceildiv(source->get_size()[0] + slice_size - 1, slice_size);
+
+    const dim3 block_size(cuda_config::warp_size,
+                          cuda_config::max_block_size / cuda_config::warp_size,
+                          1);
+    const dim3 init_grid_dim(ceildiv(result->get_stride(), block_size.x),
+                             ceildiv(num_rows, block_size.y), 1);
+
+    kernel::initialize_zero_dense<<<init_grid_dim, block_size>>>(
+        num_rows, num_cols, result->get_stride(),
+        as_cuda_type(result->get_values()));
+
+    const dim3 block_dim(slice_size, default_block_size / slice_size, 1);
```

Comment:
What if `default_block_size < slice_size`, or if `default_block_size % slice_size != 0`? Either way, the result will likely be wrong.

I would recommend using
```c++
constexpr int threads_per_row = cuda_config::warp_size; // if you want multiple threads per row
const dim3 block_dim(default_block_size, 1, 1);
const dim3 grid_dim(ceildiv(slice_size * slice_num * threads_per_row, default_block_size), 1, 1);
```
(in combination with the proposed changes above).